lib/options.nix

```
@@ -45,158 +47,262 @@ let
 in
 rec {
 
-  /* Returns true when the given argument is an option
+  /**
+    Returns true when the given argument is an option
 
-     Type: isOption :: a -> bool
+    # Inputs
+
+    `value`
+    : Any value to check whether it is an option
+
+    # Examples
+    :::{.example}
+    ## `lib.options.isOption` usage example
+
+    ```nix
+    isOption 1             // => false
+    isOption (mkOption {}) // => true
+    ```
+
+    :::
+
+    # Type
+
+    ```
+    isOption :: a -> bool
+    ```
 
-     Example:
-       isOption 1             // => false
-       isOption (mkOption {}) // => true
   */
   isOption = lib.isType "option";
 
-  /* Creates an Option attribute set. mkOption accepts an attribute set with the following keys:
+  /**
+    Creates an Option attribute set. mkOption accepts an attribute set with the following keys:
+
+    All keys default to `null` when not given.
+
+    # Inputs
+
+    Structured function argument
+    : Attribute set containing the following attributes.
```

This is incorrect. It does not need to contain any attributes.

```suggestion
    : Attribute set containing none or some of the following attributes.
```